src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -358,6 +370,40 @@ public List<int> ConsensusTipChanged(ChainedHeader newConsensusTip)
         {
             this.logger.LogTrace("({0}:'{1}')", nameof(newConsensusTip), newConsensusTip);
 
+            ChainedHeader oldConsensusTip = this.GetConsensusTip();
+            ChainedHeader fork = newConsensusTip.FindFork(oldConsensusTip);
+            ChainedHeader currentHeader = newConsensusTip;
+
+            this.logger.LogTrace("Old consensus tip: '{0}', new consensus tip: '{1}', fork point: '{2}'.", oldConsensusTip, newConsensusTip, fork);
+
+            // Consider blocks that became a part of our best chain as consumed.
+            while (currentHeader != fork)
+            {
+                this.UnconsumedBlocksDataBytes -= currentHeader.Block.BlockSize.Value;
+                this.logger.LogTrace("Size of unconsumed block data is decreased by {0}, new value is {1}.", currentHeader.Block.BlockSize.Value, this.UnconsumedBlocksDataBytes);
```

Comment:
Cant `Block` be null between the two tips?

Reply:
No. We can't delete block data ahead of our consensus tip. Only when we switch to it we call this method and later in the code we call CleanOldBlockDataFromMemory that might delete that block data. 

So at the point of usage here we have a guarantee that the block data is not null.